cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/table.spec.js

```diff
@@ -0,0 +1,188 @@
+    });
+
+    afterEach(() => {
+      cy.deleteWorkspaceByName(`${workspaceName}`);
+      cy.deleteDataSourceByName(datasourceName);
+      cy.deleteIndex('data_logs_small_time_1');
+      cy.window().then((win) => {
+        win.localStorage.clear();
+        win.sessionStorage.clear();
+      });
+    });
+
+    generateAllTestConfigurations(generateTableTestConfiguration, {
+      indexPattern: 'data_logs_small_time_1*',
+      index: 'data_logs_small_time_1',
+    }).forEach((config) => {
+      describe(`${config.testName}`, () => {
+        beforeEach(() => {
+          if (config.datasetType === 'INDEX_PATTERN') {
+            cy.createWorkspaceIndexPatterns({
+              workspaceName: workspaceName,
+              indexPattern: 'data_logs_small_time_1',
+              timefieldName: 'timestamp',
+              dataSource: datasourceName,
+              isEnhancement: true,
+            });
+          }
+          cy.navigateToWorkSpaceSpecificPage({
+            workspaceName: workspaceName,
+            page: 'discover',
+            isEnhancement: true,
+          });
+        });
+
+        it(`should allow expand multiple documents for ${config.testName}`, () => {
+          // Setup
+          cy.setDataset(config.dataset, datasourceName, config.datasetType);
+          cy.setQueryLanguage(config.language);
+          setDatePickerDatesAndSearchIfRelevant(config.language);
+          // expanding a document in the table
+          cy.get('[data-test-subj="docTableExpandToggleColumn"]')
+            .find('[type="button"]')
+            .eq(2)
+            .click();
+
+          // expanding a document in the table
+          cy.get('[data-test-subj="docTableExpandToggleColumn"]')
+            .find('[type="button"]')
+            .eq(3)
+            .click();
+
+          // checking the number of exapnded documents visible on screen
+          cy.get('[data-test-subj="tableDocViewRow-_index"]').should('have.length', 2);
+
+          // switch query should keep expanded state
+          // TODO: allow switch to other languages
+          if (config.language === QueryLanguages.DQL.name) {
+            cy.setQueryLanguage('Lucene');
+          } else if (config.language === QueryLanguages.Lucene.name) {
+            cy.setQueryLanguage('DQL');
+          } else if (config.language === QueryLanguages.SQL.name) {
+            cy.setQueryLanguage('PPL');
+          } else {
+            cy.setQueryLanguage('OpenSearch SQL');
+          }
+          cy.get('[data-test-subj="tableDocViewRow-_index"]').should('have.length', 2);
+        });
+      });
+    });
+
+    generateIndexPatternTestConfigurations(generateTableTestConfiguration, {
+      indexPattern: 'data_logs_small_no_time_1*',
+      supportedLanguages: [QueryLanguages.DQL, QueryLanguages.Lucene],
+    }).forEach((config) => {
+      describe(`${config.testName}`, () => {
+        beforeEach(() => {
+          if (config.datasetType === 'INDEX_PATTERN') {
```

Comment:
nitpick here too

Reply:
will update this with other files in a separate PR